utils/consensus/consensus.py

```diff
@@ -0,0 +1,446 @@
+        self.brake = brake
+        self.steering = steering
+        self.throttle = throttle
+        self.last_measurement = None
+
+    def reading_sleep(self, duration=1.0):
+        """
+        Used in place of `time.sleep()`. Enables "waiting" for some interval while maintaining
+        confidence that once we start unpacking at the car's reports again that data hasn't gone
+        stale.
+        """
+
+        end = time.time() + duration
+
+        while time.time() < end:
+            self.bus.recv_report()
+
+    def enable(self):
+        """
+        Enable all OSCC modules.
+        """
+
+        while True:
+
+            success = self.enable_module(self.brake)
+
+            if not success:
+                continue
+
+            success = self.enable_module(self.steering)
+
+            if not success:
+                continue
+
+            success = self.enable_module(self.throttle)
+
+            if not success:
+                continue
+
+            break
+
+    def disable(self):
+        """
+        Disable all OSCC modules.
+        """
+        self.disable_module(self.brake)
+        self.disable_module(self.steering)
+        self.disable_module(self.throttle)
+
+    def enable_module(self, module):
+        """
+        Enable a single OSCC modules. Print status, success and failure reports.
+        """
+
+        print(
+            Fore.MAGENTA + ' status: ',
+            Style.RESET_ALL,
+            'attempting to enable',
+            module.module_name,
+            'module')
+
+        # Attempt to enable the module parameter. Under the hood, this sends the enable brakes CAN
+        # frame to OSCC/DriveKit over its CAN gateway.
+        self.bus.enable_module(module)
+
+        # Verify the module parameter is enabled by listening to the OSCC/DriveKit CAN gateway for
+        # a status message that confirms it. Set the `success` flag so we can report and handle
+        # failure
+        success = self.bus.check_module_enabled_status(
+            module,
+            expect=True)
+
+        if success:
+            print(Fore.GREEN + ' success:', Style.RESET_ALL, module.module_name, 'module enabled')
+        else:
+            print(
+                Fore.RED +
+                ' error:  ',
+                Style.RESET_ALL,
+                module.module_name,
+                'module could not be enabled')
+
+        self.reading_sleep()
+
+        return success
+
+    def disable_module(self, module):
+        """
+        Disable a single OSCC modules. Print status, success and failure reports.
+        """
+
+        print(
+            Fore.MAGENTA + ' status: ',
+            Style.RESET_ALL,
+            'attempting to disable',
+            module.module_name,
+            'module')
+
+        # Attempt to enable the module parameter. Under the hood, this sends the enable brakes CAN
+        # frame to OSCC/DriveKit over its CAN gateway.
+        self.bus.enable_module(module)
```

Comment:
I think this should be disable_module, switching it to disable does yield desired behavior of disabling but speed of disable is way faster than the rest of the tests on here.

Reply:
addressed in 2e4e851

Reply:
well kind of addressed. This method should still be calling `self.bus.enable_module` as it's the `enable_module` method. In the `disable_module` method the call to `self.bus.enable_module` was replaced with the expected `self.bus.disable_module`